src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets

```diff
@@ -209,17 +219,56 @@ Copyright (c) .NET Foundation. All rights reserved.
        * Assets from the referenced packages. These will be implicitly included when nuget
          restores the package and includes the package props file for the package.
   -->
+
+    <!-- StaticWebAssets from the current project come from ResolveCurrentProjectStaticWebAssetsInputs which is a dependency of this target. -->
+    <!-- StaticWebAssets from referenced projects come from ResolveReferencedProjectsStaticWebAssets which is a dependency of this target. -->
+    <!-- StaticWebAssets from packages are already available, so we don't do anything. -->
   <Target
     Name="ResolveStaticWebAssetsInputs"
-    DependsOnTargets="$(ResolveStaticWebAssetsInputsDependsOn)">
+    DependsOnTargets="$(ResolveStaticWebAssetsInputsDependsOn)" />
 
-    <!-- StaticWebAssets from the current project -->
+  <!-- This is a helper task to compute the project references we need to invoke to retrieve
+       the static assets for a given application. We do it this way so that we can
+       pass additional build properties to compute the assets from the package when referenced
+       as a project. For example, Identity uses this hook to extend the project reference and
+       pass in the bootstrap version to use.
+  -->
+  <Target Name="_ResolveStaticWebAssetsProjectReferences"
+    DependsOnTargets="ResolveReferences">
 
-    <PropertyGroup>
-      <StaticWebAssetBasePath Condition="$(StaticWebAssetBasePath) == ''">_content/$(PackageId)</StaticWebAssetBasePath>
-    </PropertyGroup>
+    <ItemGroup>
+      <_StaticWebAssetsProjectReference
+        Include="%(ReferencePath.MSBuildSourceProjectFile)"
+        Condition="'%(ReferencePath.MSBuildSourceProjectFile)' != ''" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target Name="ResolveReferencedProjectsStaticWebAssets" DependsOnTargets="$(ResolveReferencedProjectsStaticWebAssetsDependsOn)">
+
+    <MSBuild
+      Condition="'@(_StaticWebAssetsProjectReference->Count())' != '0'"
+      Projects="@(_StaticWebAssetsProjectReference)"
+      BuildInParallel="$(BuildInParallel)"
+      ContinueOnError="!$(BuildingProject)"
+      Targets="GetCurrentProjectStaticWebAssets"
+      SkipNonexistentTargets="true">
+      <Output TaskParameter="TargetOutputs" ItemName="_ReferencedProjectStaticWebAssets" />
```

Comment:
I think there can still be double evaluation here if projects have different configurations. 

We aren’t respecting %(ProjectReference.SetXxx)

I think these need to mirror ResolveProjectReferences very carefully. @rainersigwald

Also seems that we'll br calling this on the outer build of multitargeted tfms rather than the matching tfm chosen, which seems wrong.

Reply:
Yeah, it doesn't respect solution configurations or metadata on the project references.

Reply:
@nguerrera \ @rainersigwald could you have a quick look at it and confirm if this looks correct now?